api/proto/teleport/legacy/client/proto/authservice.proto

```diff
@@ -2895,4 +2897,22 @@ service AuthService {
 
   // UpdateHeadlessAuthenticationState is a request to update a headless authentication's state.
   rpc UpdateHeadlessAuthenticationState(UpdateHeadlessAuthenticationStateRequest) returns (google.protobuf.Empty);
+
+  // CreatePlugin creates a plugin resource.
+  rpc CreatePlugin(types.PluginV1) returns (google.protobuf.Empty);
+
+  // DeletePlugins deletes a plugin resource.
+  rpc DeletePlugin(types.ResourceRequest) returns (google.protobuf.Empty);
+
+  // DeleteAllPlugins deletes all plugin resources.
+  rpc DeleteAllPlugins(google.protobuf.Empty) returns (google.protobuf.Empty);
+
+  // ListPlugins lists current plugin resources.
+  rpc ListPlugins(types.ListPluginsRequest) returns (types.ListPluginsResponse);
+
+  // GetPlugin gets a plugin resource.
+  rpc GetPlugin(types.ResourceRequest) returns (types.PluginV1);
+
+  // GetPlugins gets all Plugin resources.
+  rpc GetPlugins(google.protobuf.Empty) returns (types.PluginListV1);
```

Comment:
Is the `GetPlugins` necessary? Given that we already have the `ListPlugins`
I would say we should have only method for listing and we usually prefer the `ListXxx` one because is allows for pagination (so, less resource consumption).